src/include/txn_inline.h

```diff
@@ -338,15 +356,33 @@ __wt_txn_op_delete_commit_apply_timestamps(WT_SESSION_IMPL *session, WT_REF *ref
                 (*updp)->durable_ts = txn->durable_timestamp;
             }
     }
-    page_del = ref->page_del;
-    if (page_del != NULL && page_del->timestamp == WT_TS_NONE) {
-        page_del->timestamp = txn->commit_timestamp;
-        page_del->durable_timestamp = txn->durable_timestamp;
-    }
+
+    __wt_txn_op_delete_commit_apply_page_del_timestamp(session, ref);
 
     WT_REF_UNLOCK(ref, previous_state);
 }
 
+/*
+ * __wt_txn_should_assign_timestamp --
+ *     Updates without a commit time and logged objects don't have timestamps, and only the most
+ *     recently committed data matches files on disk.
```

Comment:
Could you improve this comment? I think it is describing the conditions where timestamps aren't applied to ops, and what the alternative is (only the most recently committed data matches)

Reply:
Fixed https://github.com/wiredtiger/wiredtiger/pull/9620/commits/ff2d37ea8db649118482cc0b7086f092b8fe72f6.